Sources/BraintreeAmericanExpress/BTAmericanExpressClient.swift

```diff
@@ -48,4 +49,23 @@ import BraintreeCore
             completion(rewardsBalance, nil)
         }
     }
+
+    ///  Gets the rewards balance associated with a Braintree nonce. Only for American Express cards.
+    /// - Parameters:
+    ///   - nonce: A nonce representing a card that will be used to look up the rewards balance.
+    ///   - currencyIsoCode: The currencyIsoCode to use. Example: 'USD'
+    /// - Returns: A `BTAmericanExpressRewardsBalance` object with information about the rewards balance
+    /// - Throws: An `Error` describing the failure
+    @objc(getRewardsBalanceForNonce:currencyIsoCode:completionHandler:)
```

Comment:
I don't think we want to expose both of these to ObjC - just the non-async await one

Reply:
The async/await ones are automatically exposed regardless because we use `@objcMembers` and `async/await` is automatically bridged to a `completionHandler`. We can certainly move away from `@objcMembers` and just use `@objc` across these classes though. Thoughts?

Reply:
> The async/await ones are automatically exposed regardless because we use @objcMembers and async/await is automatically bridged to a completionHandler. 

Interesting. So I just messed around taking a look at the auto-generated Swift header. With the current implementation, we only have one `getRewardsBalanceForNonce`, because our ObjC naming annotation matches for both functions. If the naming _didn't_ match for both, you're right that one will get auto-generated due to our `objCMembers`. This results in two `getRewardsBalanceForNonce` methods.

IMO the most explicit / lowest margin of error route here is to not use objCMemembers, so we always know exactly which function is being exposed and it's naming convention. Otherwise we'll always need to make sure that both our completionHandler and async-await functions have the same objC naming annotation - does that make sense? what do you think, Jax?

Reply:
Yeah that totally makes sense. I am also inclined to go the route of not using `@objCMembers` in these classes at least and just rely on `@objc`. It's mainly just on the clients, we don't need to nuke it everywhere. I'll make this update.

Reply:
Updated: https://github.com/braintree/braintree_ios/pull/975/commits/eb9e6ef1d1b8a87a2437a6b64b45cf30c0b7cdde